src/main/java/ru/stqa/selenium/factory/ThreadLocalSingleWebDriverPool.java

```diff
@@ -39,30 +40,24 @@ public void run() {
 
   @Override
   public WebDriver getDriver(String hub, Capabilities capabilities) {
-    String newKey = createKey(capabilities, hub);
     if (tlDriver.get() == null) {
+      return createNewDriver(capabilities, hub);
+    }
+    String key = driverToKeyMap.get(tlDriver.get());
+    if (key == null) {
+      // The driver was dismissed in a wrong way
+      return createNewDriver(capabilities, hub);
```

Comment:
I think that WebDriver return in createNewDriver() method makes code less readable and obvoius. Four (4!) return statements let the code smell.